### PR TITLE
i18n pipeline fixes; added French Canadian and other languages 

### PR DESCRIPTION
**Pull request by OmarIthawi**
# Overview
Translation tooling is now outdated starting on #1. This pull request brings it back by reverting #1 among other modifications.

I've already done some work back in 2019 on the Stanford fork but it was closed unmerged https://github.com/Stanford-Online/xblock-free-text-response/pull/42.

# Test Instructions
- Install on devstack
- Use French (Canadian): `fr-ca` and test the XBlock

# TODO
- [x] Add a test for `plurals` issue
- [x] Update translations
- [x] Add translation checks to `.travis.yml`
- [x] Pass all tests
- [x] Bump the version number in `setup.py`
- [x] Translations to work on devstack/sandbox
- [x] Attach screenshots
- [ ] Submit PR against `edx-platform` to bump the version
- [ ] Upload to PyPi

# Screenshots of Esperanto
<kbd>ftr-i18n</kbd>

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @OmarIthawi! I've created [OSPR-5299](https://openedx.atlassian.net/browse/OSPR-5299) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by OmarIthawi** [comment]
@natabene Could you please line this up for review?

**Comment by natabene** [comment]
@OmarIthawi Thank you for your contribution. Sure.

File: freetextresponse/translations/config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - eo  # Esperanto
+    - ar  # Arabic
+    - de_DE  # German
+    - es_419  # Spanish (Latin America)
+    - ja_JP  # Japanese
+    - fr  # French
+    - fr_CA  # French (Canada)
+    - he  # Hebrew
+    - hi  # Hindi
+    - ja_JP  # Japanese
```
Comment (sarina) [inline freetextresponse/translations/config.yaml:14]: This is duplicated on line 9

File: freetextresponse/translations/config.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Configuration for i18n workflow.
+
+locales:
+    - en  # English - Source Language
+    - eo  # Esperanto
```
Comment (sarina) [inline freetextresponse/translations/config.yaml:5]: if this is used in `dummy_locales` should it also be in `locales`?

File: tox.ini [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
 
 [testenv]
+usedevelop=True
 deps =
     django22: Django>=2.2,<2.3
     django30: Django>=3.0,<3.1
     -rrequirements/test.txt
 commands =
-    coverage run manage.py test
+    coverage run manage.py test  # This runs 0 tests. TODO: change to `test freetextresponse.tests`
```
Comment (sarina) [inline tox.ini:11]: if this is a no-op, why not comment it out for now and put the todo at the top level?
```
# TODO: Runs 0 tests. change to `test freetextresponse.tests`
# after migrating `test_all.py` to Python 3.
# coverage run manage.py test
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Version 1.1.2
+* Activates and fix translation tooling with Travis CI checks
+* Pull the latest translations from Transifex (to date: December 11, 2020)
```
Comment (sarina) [inline CHANGELOG.md:5]: Is it worth pulling translations again, since we're a month+ out?

**Comment by sarina** [comment]
The PR description mentions that you'll attach screenshots here

**Comment by sarina** [comment]
Overall this looks good. I'm strapped for time but will try to find some to install the XBlock and test it.

**Comment by OmarIthawi** [comment]
Thanks Sarina for the review. I'll address them and push the changes.

File: CHANGELOG.md (hunk 4 above)
Comment (OmarIthawi) [reply CHANGELOG.md:5]: Doing that now.

```suggestion
* Pull the latest translations from Transifex (to date: January 21, 2021)
```

File: freetextresponse/translations/config.yaml (hunk 1 above)
Comment (OmarIthawi) [reply freetextresponse/translations/config.yaml:14]: removed.

File: freetextresponse/translations/config.yaml (hunk 2 above)
Comment (OmarIthawi) [reply freetextresponse/translations/config.yaml:5]: not really. removed.

File: tox.ini (hunk 3 above)
Comment (OmarIthawi) [reply tox.ini:11]: done

**Comment by OmarIthawi** [comment]
All review comments have been addressed so far. I've just found a bug that the `text.mo` isn't getting picked up which makes i18n nonfunctional:

<kbd>image</kbd>

I'm marking this PR as `(WIP)` and will work to fix the issue.

**Comment by OmarIthawi** [comment]
@mahyard I've fixed the `translations` folder issue. Could you please take another look?

**Comment by OmarIthawi** [comment]
@sarina @mahyard thank you both for the review. I think this PR is ready for merge.

I've done the following:

 - Tested it in a Juniper sandbox
 - Took a screenshot
 - Squashed the commits to the minimum

**Comment by mahyard** [comment]
@OmarIthawi when I click the xblock's edit button from studio, I get this error:
image

using eo as language doesn't solve the problem. 
unfortunately, there is no clue about this in the log files.

**Comment by OmarIthawi** [comment]
> @OmarIthawi when I click the xblock's edit button from studio, I get this error:
> image
> 
> using eo as language doesn't solve the problem.
> unfortunately, there is no clue about this in the log files.

which release are you using?

**Comment by mahyard** [comment]
You are welcome @OmarIthawi,
So that's the point. we are currently on ironwood.
BTW I was able to install the xblock and render it in **eo** and **ar** languages.
I appreciate your effort

**Review by sarina — APPROVED** [review]

**Comment by sarina** [comment]
@OmarIthawi do you need me to merge this for you?

**Comment by natabene** [comment]
@sarina If you think this is fit to merge, please merge.

**Comment by edx-webhook** [comment]
@OmarIthawi 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-free-text-response&entry.867055334=https://github.com/edx/xblock-free-text-response/pull/9&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2020-12-11+08:33&entry.1917517419=2021-01-25+21:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**